### PR TITLE
KAS-ECC Component mode and some info for non-Component mode

### PR DESCRIPTION
**Pull request by bfussell**
Plus fixed a bug in TOTP password generation when there was a NULL in the string which caused the key length to be wrong.

### CODE REVIEW

File: app/app_main.c [hunk 1]
```diff
@@ -3645,23 +3670,60 @@ static ACVP_RESULT app_kas_ecc_handler(ACVP_TEST_CASE *test_case)
...
-        EC_GROUP_free(group);
-        BN_free(cx);
-        BN_free(cy);
-        printf("EC_KEY_generate_key failed\n");
-        return ACVP_CRYPTO_MODULE_FAIL;
+    if (tc->test_type == ACVP_KAS_ECC_TT_VAL) {
+        if (!BN_hex2bn(&ix, (char *)tc->pix)) {
+            EC_GROUP_free(group);
+            BN_free(cx);
+            BN_free(cy);
```
Comment (fliphil) [inline app/app_main.c]: Down to preference, but goto cleanup and then the memory can be freed there. Saves redundant lines for each for each of these error statements. Up to you :)

Comment (bfussell) [reply app/app_main.c]: I'll commit that change with the second half of the TOTP fix that I missed.
You'll have to re-approve I think.

File: app/app_main.c [hunk 2]
```diff
@@ -3761,7 +3714,12 @@ static ACVP_RESULT app_kas_ecc_handler(ACVP_TEST_CASE *test_case)
...
     }
-    OPENSSL_cleanse(Z, Zlen);
+    rv = ACVP_SUCCESS;
+
+error:
+    if (Z) {
+        OPENSSL_cleanse(Z, Zlen);
+    }
     OPENSSL_free(Z);
     EC_KEY_free(ec);
```
Comment (fliphil) [inline app/app_main.c]: Do you need to check "if (ec) EC_KEY_free(ec);" ?

Comment (bfussell) [reply app/app_main.c]: no, only reason Z needed that check was since it was a cleanse and not a free

**Comment by fliphil** [comment]
TOTP fix looks good 👍

**Pull request merged**